app/main.py

```diff
@@ -1,2 +1,86 @@
+from typing import Generator
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self._size = 8
+        self._threshold = 0.3
+        self._table: list[list] = [[] for _ in range(self._size)]
+        self._resize_const = 2
+
+    def print_table(self) -> None:
+        print(self._table)
+
+    def __getitem__(self, key: any) -> all:
+        hash_key = hash(key)
+        table_index = hash_key % self._size
+
+        for _ in range(self._size):
+            if self._table[table_index] and self._table[table_index][1] == key:
+                return self._table[table_index][0]
+            else:
+                table_index = next(self._gen_next_slot(table_index))
+                if (self._table[table_index]
+                        and self._table[table_index][1] == key):
+                    return self._table[table_index][0]
```

Comment:
The logic for handling collisions in the `__getitem__` method is not correctly implemented. The loop should continue searching for the key until it either finds the key or determines that the key is not present. The current implementation may not correctly handle cases where the key is not found in the first slot checked.